lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1191 @@
+    q = False
+
+    def __init__(self, **kwargs):
+        my_mod = kwargs.get('module')
+        if my_mod:
+            self.module = my_mod
+        else:
+            self.module = the_logger.default_module
+
+        my_level = kwargs.get('log_level')
+        if my_level:
+            self.level = my_level
+        else:
+            self.level = the_logger.default_log_level
+
+    def config(self, my_mod):
+        self.module = my_mod
+
+    def log(self, level, message):
+        if HAS_Q and the_logger.q:
+            q("log message: " + message)
+        if self.level > level:
+            return
+        if self.module is None:
+            print_(message, file=sys.stderr)
+        else:
+            self.module.log(message)
+
+    def log_print(self, level, *args, **kwargs):
+        if HAS_Q and the_logger.q:
+            q("log print args: " + " ".join(args) + " " + repr(kwargs))
+        if self.level > level:
+            return
+        if self.module is None:
+            print_(args, file=sys.stderr, **kwargs)
+        else:
+            self.module.log()
+
+
+main_logger = the_logger()
+
+
+def await_resource(conn, instance_id, status, module, await_pending=None):
+    """wait for a resource to change into a given status
+
+    Await resource repeatedly polls an RDS waiting for it to change
+    into a given status or disappear.
+
+    N.B. If the resource disappears it will return None and this mightn
+    happen while awaiting _any_ state.
+    """
+    wait_timeout = module.params.get('wait_timeout') + time.time()
+    # Refresh the resource immediately in case we just changed it's state;
+    # should we sleep first?
+    assert instance_id is not None
+    resource = get_db_instance(conn, instance_id)
+    main_logger.log(50, "await_resource called for " + instance_id)
+    main_logger.log(40, "wait is {0} {1} await_pending is {2} status is {3}".format(
+        str(wait_timeout), str(time.time()), str(await_pending), str(resource['DBInstanceStatus'])))
+    rdat = resource["PendingModifiedValues"]
+    while ((await_pending and rdat) or resource['DBInstanceStatus'] != status) and wait_timeout > time.time():
+        main_logger.log(70, "waiting with resource{}  ".format(str(resource)))
+        time.sleep(5)
+        # Temporary until all the rds2 commands have their responses parsed
+        current_id = resource.get('DBInstanceIdentifier')
+        if current_id is None:
+            module.fail_json(
+                msg="There was a problem waiting for RDS instance %s" % resource.instance)
+        resource = get_db_instance(conn, current_id)
+        if resource is None:
+            break
+        rdat = resource["PendingModifiedValues"]
+    # resource will be none if it has actually been removed - e.g. we were waiting for deleted
+    # status; maybe that should be an error in other situations?
+    if wait_timeout <= time.time() and resource is not None and resource['DBInstanceStatus'] != status:
```

Comment:
Instead of checking times at the `while` control statement *and* here, just move this into the loop and have it fail from there.